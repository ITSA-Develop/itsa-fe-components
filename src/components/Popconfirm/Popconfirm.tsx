import { Popconfirm as AntPopconfirm, PopconfirmProps } from 'antd';

export const Popconfirm = ({ ...rest }: PopconfirmProps) => {
	return <AntPopconfirm {...rest} />;
};
