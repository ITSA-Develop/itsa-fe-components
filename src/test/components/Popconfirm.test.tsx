import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { vi } from 'vitest';
import { Popconfirm } from '../../components/Popconfirm/Popconfirm';

describe('Popconfirm component', () => {
	it('renders the Popconfirm and shows confirmation text on click', async () => {
		const confirmText = 'Are you sure?';
		const onConfirm = vi.fn();

		const { container } = render(
			<Popconfirm title={confirmText} onConfirm={onConfirm}>
				<button>Delete</button>
			</Popconfirm>,
		);

		const button = screen.getByText('Delete');
		fireEvent.click(button);

		await waitFor(() => {
			const confirmation = screen.getByText(confirmText);
			expect(confirmation).toBeVisible();
		});

		const confirmBtn = screen.getByText('OK');
		fireEvent.click(confirmBtn);

		expect(onConfirm).toHaveBeenCalled();
		expect(container).toMatchSnapshot();
	});
});
