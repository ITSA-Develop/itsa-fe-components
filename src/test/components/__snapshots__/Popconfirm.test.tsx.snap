// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popconfirm component > renders the Popconfirm and shows confirmation text on click 1`] = `
<div>
  <button
    aria-describedby="test-id"
  >
    Delete
  </button>
</div>
`;
